packages/common/src/core/types/DynamicValues.ts

```diff
@@ -0,0 +1,102 @@
+import { ISearchOptions } from "@esri/arcgis-rest-portal";
+import { IConfigurationSchema } from "../schemas";
+
+/**
+ * The source for a dyanmic value. Used to determine how to resolve the value
+ */
+export type DynamicValueSource = "item-query" | "service-query" | "portal";
+
+/**
+ * Subset of the ISearchOptions interface, used for item queries
+ */
+export type ItemSearchOptions = Partial<Pick<ISearchOptions, "q" | "filter">>;
+
+/**
+ * Options used to defined a feature service query
+ */
+export interface IServiceSearchOptions {
```

Comment:
```suggestion
export interface IServiceQueryOptions {
```

also, I wonder if we want to use [IStatisticDefinition](https://github.com/Esri/arcgis-rest-js/blob/3.x/packages/arcgis-rest-types/src/statisticDefinition.ts) here, or at least sync up on the aggregation types.

Reply:
I like the consistency on "query". 

I looked at using IStatsDefinition, and went back and forth on it. The aggs are a subset, and in rest-js they are not their own type... it's just defined inline so I can't derive other types directly from it. 

I left out a few though - we can def add them in but they require more props to be passed, and return a different structure - def something to discuss if it's needed for MVP